common/src/main/java/com/wynntils/models/raid/RaidModel.java

```diff
@@ -151,9 +208,151 @@ public void onWorldStateChange(WorldStateEvent event) {
         }
     }
 
+    // Process raid rewards
+    @SubscribeEvent
+    public void onScreenInit(ScreenInitEvent.Pre e) {
```

Comment:
I'm not sure this is the best way to achieve what you want. It feels very intrusive. It's long past bedtime and I'm just glancing through this, so I don't have any other suggestions handy, though.

Reply:
I think lootrun model does the same, we likely can't do anything better.

Reply:
Hm. I'm pretty sure I wrote some kind of class that could filter through a container using callback/lambda to check items. I thought I used it for some emerald handling stuff. I don't have my IDE handy now, and I can't find it in the github source view though...

Reply:
I was thinking about the InventoryWatcher. So it is not directly usable here. Nevertheless, I believe it would be good to separate this logic in two parts -- one that does the handling of looking for the container opening/closing and register items being set, and one that does what it now is we want to have done for this specific case. Especially if we already do something like this for Lootrun.

I will not set this as an absolute requirement for merging this PR, though. In fact, it would probably be better to extract out the logic from Lootrun first, and then merge in that new logic here and re-use it. But if this is too much work, I can turn a blind eye towards this.